src/sync/impl/sync_client.hpp

```diff
@@ -39,7 +39,7 @@ namespace _impl {
 using ReconnectMode = sync::Client::ReconnectMode;
 
 struct SyncClient {
-    SyncClient(std::unique_ptr<util::Logger> logger, SyncClientConfig const& config)
+    SyncClient(std::unique_ptr<util::Logger> logger, SyncClientConfig const& config, std::shared_ptr<const SyncManager> sync_manager)
```

Comment:
It is used if `NETWORK_REACHABILITY_AVAILABLE` is set.

Reply:
Okay, fair enough, but can we add an else statement with `(void) sync_manager` then to avoid triggering the warning when building without that flag? (I think it is an iOS only flag?)

Reply:
Done